test/Table_Tests/src/Database/SQLite_Spec.enso

```diff
@@ -252,53 +335,86 @@ sqlite_spec connection prefix =
          the missing statistics.
     aggregate_selection = Common_Table_Operations.Aggregate_Spec.Test_Selection.Config advanced_stats=False text_shortest_longest=False first_last=False first_last_row_order=False multi_distinct=False aggregation_problems=False nan=False date_support=False
     agg_in_memory_table = (enso_project.data / "data.csv") . read
-    agg_table = agg_in_memory_table.select_into_database_table connection (Name_Generator.random_name "Agg1") primary_key=Nothing temporary=True
-    empty_agg_table = (agg_in_memory_table.take (First 0)).select_into_database_table connection (Name_Generator.random_name "Agg_Empty") primary_key=Nothing temporary=True
 
-    setup = Common_Table_Operations.Main.Test_Setup.Config prefix agg_table empty_agg_table table_builder materialize is_database=True test_selection=common_selection aggregate_test_selection=aggregate_selection connection=connection
-    sqlite_specific_spec prefix connection setup
-    Common_Table_Operations.Main.spec setup
+    agg_table_fn = _ ->
+        connection = create_connection_func Nothing
+        agg_in_memory_table.select_into_database_table connection (Name_Generator.random_name "Agg1") primary_key=Nothing temporary=True
+
+    empty_agg_table_fn = _ ->
+        connection = create_connection_func Nothing
+        (agg_in_memory_table.take (First 0)).select_into_database_table connection (Name_Generator.random_name "Agg_Empty") primary_key=Nothing temporary=True
+
+    setup = Common_Table_Operations.Main.Test_Setup.Config prefix agg_table_fn empty_agg_table_fn table_builder materialize is_database=True test_selection=common_selection aggregate_test_selection=aggregate_selection create_connection_func=create_connection_func
+    sqlite_specific_spec suite_builder prefix create_connection_func setup
+    Common_Table_Operations.Main.add_specs suite_builder setup
 
-    connection.close
 
 ## PRIVATE
 supported_replace_params : Set Replace_Params
 supported_replace_params =
     e = [Replace_Params.Value Text Case_Sensitivity.Default False, Replace_Params.Value Text Case_Sensitivity.Sensitive False, Replace_Params.Value Text Case_Sensitivity.Default True, Replace_Params.Value Text Case_Sensitivity.Sensitive True, Replace_Params.Value Text Case_Sensitivity.Insensitive True]
     Set.from_vector e
 
-spec =
-    enso_project.data.create_directory
-    file = enso_project.data / "transient" / "sqlite_test.db"
-    file.delete_if_exists
+backing_file =
+    transient_dir = enso_project.data / "transient"
+    assert transient_dir.exists ("There should be .gitignore file in the " + transient_dir.path + " directory")
+    transient_dir / "sqlite_test.db"
```

Comment:
The assert message is only vaguely connected to the condition it is checking, that is quite confusing for me.

Can we align this?

The assert **does not** check for a `.gitignore` file, but for the directory existing.
I know the existence of `.gitignore` will imply existence of the parent dir, but the converse is not true. Let's either explicitly check for `.gitignore` OR reword the condition. Otherwise it is just confusing.